src/cogent3/core/annotation.py

```diff
@@ -172,11 +173,45 @@ def detach_annotations(self, annots):
     def add_feature(self, type, name, spans):
         return self.add_annotation(Feature, type, name, spans)
 
-    def add_variable(self, type, name, xxy_list):
-        return self.add_annotation(Variable, type, name, xxy_list)
+    def copy_to_seq(self, seq):
+        """creates a new annotation with identical content on a new sequence
 
-    def add_simple_variable(self, type, name, data):
-        return self.add_annotation(SimpleVariable, type, name, data)
+        Parameters
+        ----------
+        seq : Sequence
+            represents the new sequence.
+
+        Returns
+        -------
+        a new annotation
+        """
+        assert len(seq) == len(self.parent)
+        serialisable = copy.deepcopy(self._serialisable)
+        if isinstance(self, AnnotatableFeature):
```

Comment:
All these `ifinstance()` conditionals could possibly be replaced with the following:
```python
serialisable["parent"] = seq
new = self.__class__(**serialisable)
```